examples/README.md

```diff
@@ -0,0 +1,61 @@
+Creating a basic S2I builder image  
+---
+
+### Getting started  
+
+Directory skeleton:  
+
+* `Dockerfile` – standard Dockerfile where we’ll define the base builder image.  
+* `assemble` - responsible for building the application.  
+* `run` - responsible for running the application.  
+* `save-artifacts` - optional script for incremental builds that save built artifacts.  
+* `usage` - optional script responsible for printing the usage of the builder.  
+
+The first step that we need to do is create a `Dockerfile` that installs all the necessary tools and libraries that are needed to build and run our application.  
+You can find an example of the Dockerfile [`here`](nginx-app/Dockerfile).   
+
+The next step is to create an `assemble` script that will install our application. This is where we build python modules or bundle install our required components. We can also restore any saved artifacts from the previous image. In our example that we have [`here`](nginx-app/assemble) it will only copy our index.html over the default one.  
+
+Now we can create our `run` script that will start the application. You can see an example  [`here`](nginx-app/run).
+
+
+Optionally we can also specify a `save-artifacts` script, which allows a new build to reuses content from a previous version of the application image.
+An example can be found [`here`](nginx-app/save-artifacts).  
+
+We can also inform the user on how to build the image via the `usage` script. An example can be found [`here`](nginx-app/usage).
+```
+s2i usage nginx-app
+
+This is the nginx-centos7 S2I image:
+To use it, install S2I: https://github.com/openshift/source-to-image
+
+Sample invocation:
+
+s2i build test/ nginx-centos7 nginx-app
+
+You can then run the resulting image via:
+docker run -d -p 8080:8080 nginx-app
+and see the test via http://localhost:8080
+```
+
+The next step is to create the builder image. In the nginx-app directory issue `docker build -t nginx-centos7 .`  
+This will create a builder image from the current Dockerfile.
+
```

Comment:
right, you wanted the usage against the builder image since that is the base for our application image.